spec/requests/contact_embed_spec.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+RSpec.describe "Contact Embed" do
+  let(:edition) do
+    body_field = build(:field, id: "body", type: "govspeak")
+    document_type = build(:document_type, contents: [body_field])
+    create(:edition, document_type_id: document_type.id)
+  end
+
+  describe "GET /documents/:document/contact-embed" do
```

Comment:
Are we missing a test here that verified the behaviour of the `new` action? I'm thinking about the modal rendering context in particular. I notice it's still covered via the feature test, so I don't think it's a major concern. More my curiosity about the apparent testing gap in this particular file.

Reply:
Sure, I see what you mean. I'd considered the layout to be an aspect of config really and have only used it as a consideration when it impacts the actual view or has logic around it, rather than the less impactful layout specifier.

It's not actually something covered by feature tests, whether we render in a modal context or not the tests will still pass as it will just render the full HTML in the modal and still find the key elements. I tested this by removing the line about layout and found the full test suite succeeded. So request tests would probably be the right layer to actually test it.

Do you think we'd benefit from testing it or that it's a glaring inconsistency?

We can probably put tests for it but I'm not sure of the best mechanism to assert on it as it's essentially the absence of key HTML elements which feels the more risky negative type of assertion than something positive (unless I'm missing something we can hook onto)

I guess if we are to do it we'd be doing it in a number of places so it'd be good to have a particular expectation for it.

Reply:
I was thinking the same thing. With the feature tests, I meant that the setting of the rendering context doesn't cause an issue in either case. It could be that the modal context triggers a buggy behaviour in the associated view.

The main oddity for me was the lack of a success case test for the GET action, which made the test seem incomplete. Are we aiming for completeness in terms of what the controller is doing, since we don't have any controller unit tests?

With the layout context specifically, we could consider using the shared example approach. However, it doesn't actually inhibit the feature, like you say, so perhaps it's not worth testing until it actually causes a problem.

Reply:
Oh I'm sorry - I was on a different wavelength.

Yes I think anytime we test an endpoint for negative scenarios there should be a pairing a positive scenario, even if it's pretty trivial just, as you say, for consistency in the test. I missed this one here - my bad 😞. I've added it now and also caught that it wasn't testing Publishing API down on the post request.

Reply:
Thanks - that's much better!